test/foundry/forking/REP-002/UnattachBridgeForkTest.t.sol

```diff
@@ -54,10 +123,38 @@ contract UnattachBridgeForkTest is RoninTest {
     RoninValidatorSet(payable(address(RONIN_VALIDATOR_SET_CONTRACT))).currentPeriod();
   }
 
-  function test_Fork_UpgradeToNewImpl_WhenPeriodEnded() external onWhichFork(_roninFork) {
+  function test_Fork_UpgradeToNewImpl_WhenPeriodEnded(uint16 seed) external onWhichFork(_roninFork) {
+    vm.assume(seed != 0);
     _upgradeToVersionSwitcher();
 
-    address coinbase = block.coinbase;
+    vm.expectEmit(address(RONIN_VALIDATOR_SET_CONTRACT));
+    emit Upgraded(_newImpl);
+
+    _fastForwardToNextDay();
+    _wrapUpEpoch();
+
+    assertEq(_getProxyImplementation(RONIN_VALIDATOR_SET_CONTRACT), _newImpl);
+    assertEq(
+      _getProxyImplementation(_roninTrustedOrgProxy),
+      NotifiedMigrator(payable(_roninTrustedOrgSwitcher)).NEW_IMPL()
+    );
+    assertEq(_getProxyImplementation(_stakingProxy), NotifiedMigrator(payable(_stakingSwitcher)).NEW_IMPL());
+    assertEq(
+      _getProxyImplementation(_slashIndicatorProxy),
+      NotifiedMigrator(payable(_slashIndicatorSwitcher)).NEW_IMPL()
+    );
+
+    _updateDuplicatedTrustedOrg(seed);
+    _applyValidatorCandidate(seed);
+
+    vm.warp(block.timestamp + 3 seconds);
+    vm.roll(block.number + 1);
+
+    _fastForwardToNextDay();
+    _wrapUpEpoch();
```

Comment:
Please assert the method calls emit events as expected.

Reply:
added.